SourceDetection/Footprints.ipynb

```diff
@@ -0,0 +1,1207 @@
+   "outputs": [],
+   "source": [
+    "plt.figure()\n",
+    "afw_display = afwDisplay.Display()\n",
+    "afw_display.setMaskTransparency(100)\n",
+    "afw_display.scale('asinh', 'zscale')\n",
+    "afw_display.mtv(calexp.getMaskedImage()[2500:2900,2800:3200])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As advertised, we swapped out the data in our footprints with noise. As an aside, if you look closely you can see an 'edge' in the data where the parent object used to be, suggesting perhaps the faint tails of the eliptical's surface brightness profile were missed in detection and now remain in our noise image. \n",
+    "\n",
+    "Now we will draw some more inspiration from [sfm.py](https://github.com/lsst/meas_base/blob/master/python/lsst/meas/base/sfm.py#L353) where we will insert our parent footprint of interest back into the image using its ID from the source catalog."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "noiseReplacer.insertSource(1010357503384)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure()\n",
+    "afw_display = afwDisplay.Display()\n",
+    "afw_display.setMaskTransparency(100)\n",
+    "afw_display.scale('asinh', 'zscale')\n",
+    "afw_display.mtv(calexp.getMaskedImage()[2500:2900,2800:3200])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We have used our `NoiseReplacer` instance to ensure this parent footprint is the only source in the image, and all other's have been replaced with noise. Measurement algorithms would be ran on the image in this setting to obtain measurements for this source ID. After that, it will be replaced again with noise, and its deblended children footprints will be exchanged to have their measurements taken in turn. In the next cell we will make a small animation to let us visualize this."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig = plt.figure()\n",
+    "# ims is a list of lists, each row is a list of artists to draw in the\n",
+    "# current frame; here we are just animating one artist, the image, in\n",
+    "# each frame\n",
+    "ims = []\n",
+    "\n",
+    "src_ids = [1010357503384] +  children.tolist()\n",
+    "\n",
+    "for i in src_ids:\n",
+    "    noiseReplacer.insertSource(i)\n",
+    "    data = calexp.getMaskedImage()[2600:2800,2900:3100].getImage().array\n",
+    "    im = plt.imshow(data, origin='lower', animated=True, vmin=-100, vmax=100)\n",
+    "    ims.append([im])\n",
+    "    noiseReplacer.removeSource(i)\n",
+    "\n",
+    "HTML(animation.ArtistAnimation(fig, ims, interval=1000, blit=True,\n",
+    "                                repeat_delay=1000).to_jshtml())\n",
+    "# takes a minute to get going, "
```

Comment:
It looks like this is outputting a spurious second image. I'm not sure how to get rid of this (maybe an artifact of imshow)?

Reply:
This is what I see...

![image](https://user-images.githubusercontent.com/5605123/54051280-058d4900-41a7-11e9-9c22-f99d6112ca21.png)

Reply:
should be fixed now